Index: nss-3.34/nss/lib/ckfw/pem/ckpemver.c
===================================================================
--- nss-3.34.orig/nss/lib/ckfw/pem/ckpemver.c
+++ nss-3.34/nss/lib/ckfw/pem/ckpemver.c
@@ -53,7 +53,7 @@
  */
 const char __nss_ckpem_rcsid[] = "$Header: NSS Access to Flat Files in PEM format"
         NSS_CKPEM_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__ " $";
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information" " $";
 const char __nss_ckcapi_sccsid[] = "@(#)NSS Access to Flag Files in PEM format "
         NSS_CKPEM_LIBRARY_VERSION _DEBUG_STRING
-        "  " __DATE__ " " __TIME__;
+        "  " "Built in OBS" " " "see rpm -q --info nss for more information";
