--- nss/lib/sysinit/nsssysinit.c	2018-12-19 16:54:54.326850754 +0200
+++ nss/lib/sysinit/nsssysinit.c	2018-12-19 17:05:34.932687833 +0200
@@ -231,9 +231,19 @@
     sysdb = getSystemDB();
     userdb = getUserDB();
 
-    /* Don't open root's user DB */
+    /* return a list of databases to open. First the system database */
+    if (sysdb) {
+        const char *readonly = userCanModifySystemDB() ? "" : "flags=readonly";
+        module_list[next++] = PR_smprintf(
+           "library= "
+           "module=\"NSS system database\" "
+           "parameters=\"configdir='sql:%s' tokenDescription='NSS system database' %s\" "
+           "NSS=\"trustOrder=80 %sflags=internal,critical\"",
+            sysdb, readonly, nssflags);
+    }
+
+    /* Next the user database, but not for root. */
     if (userdb != NULL && !userIsRoot()) {
-        /* return a list of databases to open. First the user Database */
         module_list[next++] = PR_smprintf(
             "library= "
             "module=\"NSS User database\" "
@@ -252,17 +262,6 @@
             userdb, stripped_parameters);
     }
 
-    /* now the system database (always read only unless it's root) */
-    if (sysdb) {
-        const char *readonly = userCanModifySystemDB() ? "" : "flags=readonly";
-        module_list[next++] = PR_smprintf(
-            "library= "
-            "module=\"NSS system database\" "
-            "parameters=\"configdir='sql:%s' tokenDescription='NSS system database' %s\" "
-            "NSS=\"trustOrder=80 %sflags=internal,critical\"",
-            sysdb, readonly, nssflags);
-    }
-
     /* that was the last module */
     module_list[next] = 0;
 
