From 44c4c045cb725643b4fd4a4bd6c049a8797687b1 Mon, 4 Mar 2019 17:53:46 +0100
From: Andrew Branson <andrew.branson@jollamobile.com>
Date: Mon, 4 Mar 2019 16:17:04 +0100
Subject: [PATCH] Rename sqlite library to avoid clashing with system

diff --git a/nss/coreconf/config.gypi b/nss/coreconf/config.gypi
index 5813787..f1b79ba 100644
--- a/nss/coreconf/config.gypi
+++ b/nss/coreconf/config.gypi
@@ -81,7 +81,7 @@
     'nspr_lib_dir%': '<(nspr_lib_dir)',
     'nspr_include_dir%': '<(nspr_include_dir)',
     'use_system_sqlite%': '<(use_system_sqlite)',
-    'sqlite_libs%': ['-lsqlite3'],
+    'sqlite_libs%': ['-lnsssqlite3'],
     'dll_prefix': '<(dll_prefix)',
     'dll_suffix': '<(dll_suffix)',
     'freebl_name': '<(freebl_name)',
diff --git a/nss/coreconf/location.mk b/nss/coreconf/location.mk
index b11558a..943122d 100644
--- a/nss/coreconf/location.mk
+++ b/nss/coreconf/location.mk
@@ -72,7 +72,7 @@
 endif
 
 ifndef SQLITE_LIB_NAME
-    SQLITE_LIB_NAME = sqlite3
+    SQLITE_LIB_NAME = nsssqlite3
 endif
 
 MK_LOCATION = included
diff --git a/nss/lib/sqlite/manifest.mn b/nss/lib/sqlite/manifest.mn
index 022749b..b2b5191 100644
--- a/nss/lib/sqlite/manifest.mn
+++ b/nss/lib/sqlite/manifest.mn
@@ -6,7 +6,7 @@
 
 MODULE = nss
 
-LIBRARY_NAME = sqlite
+LIBRARY_NAME = nsssqlite
 LIBRARY_VERSION = 3
 MAPFILE = $(OBJDIR)/sqlite.def
 DEFINES += -DSQLITE_THREADSAFE=1
diff --git a/nss/lib/sqlite/nsssqlite.def b/nss/lib/sqlite/nsssqlite.def
new file mode 120000
index 0000000..f8cd2b2
--- /dev/null
+++ b/nss/lib/sqlite/nsssqlite.def
@@ -0,0 +1 @@
+sqlite.def
\ No newline at end of file
diff --git a/nss/lib/sqlite/sqlite.def b/nss/lib/sqlite/sqlite.def
index 00fa623..3de3051 100644
--- a/nss/lib/sqlite/sqlite.def
+++ b/nss/lib/sqlite/sqlite.def
@@ -17,7 +17,7 @@
 ;+#   directives are hidden behind ";", ";+", and ";-"
 ;+SQLITE_3 {       
 ;+    global:
-LIBRARY sqlite3 ;-
+LIBRARY nsssqlite3 ;-
 EXPORTS ;-
 sqlite3_aggregate_context;
 sqlite3_aggregate_count;
diff --git a/nss/lib/sqlite/sqlite.gyp b/nss/lib/sqlite/sqlite.gyp
index 6969893..a4c7cad 100644
--- a/nss/lib/sqlite/sqlite.gyp
+++ b/nss/lib/sqlite/sqlite.gyp
@@ -27,7 +27,7 @@
           ]
         },
         {
-          'target_name': 'sqlite3',
+          'target_name': 'nsssqlite3',
           'type': 'shared_library',
           'dependencies': [
             'sqlite'
diff --git a/nss/nss.gyp b/nss/nss.gyp
index 3ec33fd..17ddb59 100644
--- a/nss/nss.gyp
+++ b/nss/nss.gyp
@@ -19,7 +19,7 @@
           'dependencies': [
             'lib/nss/nss.gyp:nss3',
             'lib/smime/smime.gyp:smime3',
-            'lib/sqlite/sqlite.gyp:sqlite3',
+            'lib/sqlite/sqlite.gyp:nsssqlite3',
             'lib/ssl/ssl.gyp:ssl3',
             'lib/util/util.gyp:nssutil3',
           ],
